plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -1363,6 +1425,61 @@ public function set_cart_hash( $value ) {
 		$this->set_prop( 'cart_hash', $value );
 	}
 
+	/**
+	 * Stores information about whether stock was reduced.
+	 *
+	 * @param bool|string $value True if stock was reduced, false if not.
+	 *
+	 * @return void
+	 */
+	public function set_order_stock_reduced( $value ) {
+		$this->set_prop( 'order_stock_reduced', wc_bool_to_string( $value ) );
```

Comment:
These applies to all the new properties, but I'm adding the comment here so that we discuss just once:

I think it's a bit confusing that internally we are storing these properties as 'yes' or 'no' values when we really mean true/false (as that's what the getters either in the order or the datastore return).

I understand this was handy when these props were stored in metadata but we're no longer doing that in COT, and it also seems like a datastore-specific thing (whether the datastore uses yes/no or a column or whatever) and not necessarily how the property should live internally in the `WC_Order` object, which to me should be true/false.

I still think we should allow using "yes" or "no" in the setters, though, and specifically the CPT datastore could convert to yes/no when saving (in the `update_post_meta()` method) and convert from yes/no to bool when reading (`read_order_data()`).

I understand this adds a bit of complexity to the CPT implementation for little gain, but it felt more "correct" to me. In any case, this is just my opinion, so I clearly wanted to discuss with you.
Also, definitely not a blocker as everything can work with this as-is (COT automatically does the conversion when formatting values and CPT is already using yes/no) *and* we're talking about how this is internally stored in the order object, not a public interface. In fact, the public interface already converts the internal value to bool.

Reply:
I think this problem originated because anyone can do any of `true`, `yes`, `1` (and their opposites) to save data as you point out. This is not a problem in HPOS anymore because the column has bool type, so it's enforced correctly. 

CPT already has the implementation that you suggest, unless I missed something.

Reply:
Added storing in order props as bool here https://github.com/woocommerce/woocommerce/pull/34627/commits/c302a85a7ef31e9258e77341629164005d5fe17c